pkg/chains/chains.go

```diff
@@ -2,123 +2,232 @@ package chains
 
 import "fmt"
 
-// Zeta chains
+/*
+		Chain represents a blockchain network with its unique chain ID
+		ChainName is the name of the chain
+		ChainId is the unique identifier of the chain
+		Network is the network type of the chain , this can be ZETA, ETH, BSC, BTC, POLYGON
+		NetworkType is the network type of the chain, this can be MAINNET, TESTNET, DEVNET, PRIVNET
+		Vm is the virtual machine type of the chain to support smart contracts, this can be EVM, NO_VM
+		Consensus is the consensus algorithm used by the chain, this can be Tendermint, Ethereum, Bitcoin
+		IsExternal is a boolean value to determine if the chain is external to Zeta
+		IsHeaderSupported is a boolean value to determine if the chain supports headers
 
-func ZetaChainMainnet() Chain {
-	return Chain{
-		ChainName: ChainName_zeta_mainnet,
-		ChainId:   7000,
-	}
-}
+       Note ChainName is normally NetworkName + NetworkType,but in some cases the value of NetworkName + NetworkType is not unique.This is true for chains which have been deprecated or have been renamed.
+       Such as GoerliChain and MumbaiChain which have been replaced by SepoliaChain and AmoyChain respectively.
+*/
 
-func ZetaTestnetChain() Chain {
-	return Chain{
-		ChainName: ChainName_zeta_testnet,
-		ChainId:   7001,
-	}
-}
-
-func ZetaMocknetChain() Chain {
-	return Chain{
-		ChainName: ChainName_zeta_mainnet,
-		ChainId:   70000,
-	}
-}
-
-func ZetaPrivnetChain() Chain {
+// Mainnet chains
+func ZetaChainMainnet() Chain {
```

Comment:
Do you have context on the reasoning why we used functions instead of variables for chains?

I would think we could use:
```
var (
  Ethereum = Chain{}
  BSC = Chain{}
  Bitcoin = Chain{}
...
)
```

Reply:
I am not sure , I agree using variables makes more sense visually

Reply:
61caa74ed36219bc9bcfe8c74d79ce83e0d41dd6